llrt_core/src/modules/http/url.rs

```diff
@@ -470,29 +337,70 @@ pub fn file_url_to_path<'js>(ctx: Ctx<'js>, url: Value<'js>) -> Result<String> {
 }
 
 pub fn url_format<'js>(url: Class<'js, URL<'js>>, options: Opt<Value<'js>>) -> Result<String> {
-    let mut fragment = true;
-    let mut unicode = false;
-    let mut auth = true;
-    let mut search = true;
+    let mut include_fragment = true;
+    let mut unicode_encode = false;
+    let mut include_auth = true;
+    let mut include_search = true;
 
     // Parse options if provided
     if let Some(options) = options.0 {
-        if options.is_object() {
-            let options = options.as_object().unwrap();
-            if let Some(value) = options.get("fragment")? {
-                fragment = value;
+        if let Some(options) = options.as_object() {
+            if let Ok(value) = options.get("fragment") {
+                include_fragment = value;
             }
             if let Ok(value) = options.get("unicode") {
-                unicode = value;
+                unicode_encode = value;
             }
             if let Ok(value) = options.get("auth") {
-                auth = value;
+                include_auth = value;
             }
             if let Ok(value) = options.get("search") {
-                search = value
+                include_search = value
             }
         }
     }
 
-    Ok(url.borrow().format(auth, fragment, search, unicode))
+    let url = url.borrow();
```

Comment:
Maybe we should use a single string and append if conditions are met instead of joining multiple strings at the end (as they are even empty one’s being joined)

Reply:
Yeah, I like that a lot better.